New: Lists add tags to existing movies

scphantm opened the pull request:
#### Database Migration
YES - 243

#### Description
This request gives the import lists system the ability to retroactively apply tags to existing movies in the database.

#### Screenshot (if UI related)

#### Todos
- [x] Tests
- [x] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [x] [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #XXXX

Commit:
Add retroactive tag application to import lists

This commit introduces a new feature allowing tags to be retroactively applied to movies already in the database. Changes include database migration for the new column, updates to the frontend to manage this feature, and backend logic to handle retroactive tagging.

Commit:
Update localization file with retroactive tagging translation

Moved and added translations for retroactive tag application. Corrected minor organizational issues in the JSON file to maintain consistency.

scphantm commented:
I got this working.  Still not sure how to unit test it, and need to keep looking to see if i missed any documentation

This addresses issue #10507

scphantm commented:
Fixes #5299

bakerboy448 commented on src/Directory.Build.props:
Should be separate commit / better separate PR

scphantm committed:
Update Directory.Build.props

moving to another PR

scphantm committed:
Update Directory.Build.props

configured to auto-detect if building on an ARM mac or not

scphantm commented:
It doesn't look like there are any wiki pages that would need to be updated.  please correct if i am wrong, i added the columns to the two modal screens.

scphantm committed:
Update ImportList.ts

scphantm committed:
Add RetroApplyTags column to additional tables

Enhanced the migration script to include the RetroApplyTags column for the Notifications and Indexers tables, ensuring consistent data structure across related tables. This column is added with a default value of 0, similar to the ImportLists table.

scphantm committed:
Add RetroApplyTags column to multiple tables

Changed the RetroApplyTags column type in ImportLists to Boolean. Added RetroApplyTags column to Notifications, Indexers, and DownloadClients with a default value of false. This ensures consistent behavior across these tables.

scphantm committed:
Merge remote-tracking branch 'origin/feature/arm-mac-selections' into feature/retro_add_tags

scphantm commented:
Yea I'm rethinking this solution a little bit.  I hate adding those 3 additional columns just cuz I added a field.   I'm going to look at the interfaces harder and see if I can come up with a better way.

scphantm committed:
Remove RetroApplyTags from multiple tables

Removed the RetroApplyTags field from ProviderDefinition, Notifications, Indexers, DownloadClients, and relevant API resources. RetroApplyTags is now exclusively managed within ImportListDefinition.

scphantm committed:
Remove RetroApplyTags property from ProviderResource.

The RetroApplyTags property has been removed to streamline the ProviderResource class. This change will help to reduce unnecessary attributes, simplifying the overall codebase.

scphantm committed:
Add retroApplyTags boolean to openapi.json

This change introduces a new boolean field called retroApplyTags to the Radarr API V3. It ensures that the API schema includes this property for any relevant operations. This helps in handling retroactive tag application within the system.

RobinDadswell commented on src/Radarr.Api.V3/openapi.json:
```diff
@@ -9818,6 +9818,9 @@
             },
             "nullable": true
           },
+          "retroApplyTags": {
+            "type": "boolean"
+          },
```
This is autogenerated after the PR is merged so not needed as part of this

scphantm committed:
Remove retroApplyTags from openapi.json

The retroApplyTags property was deleted to streamline the API schema. This change ensures the JSON specification is more concise and easier to maintain.

scphantm commented:
What else needs completed to merge this?

brrusselburg commented:
Seconding to whoever needs to merge this that I am also eyeing it 👀

scphantm commented:
I really would like this put in.  I think its a far more popular of a feature than you realize.  I also would like to reengineer the code for the search to allow for an advanced search option that would allow you to use all the parameters available in the tmdb search database to create more advanced lists.  I would like this in before i spend the time on that code.  What needs to happen to get this merged?  I will fix the conflict real quick but please give some guidance here, i would like to contribute to the darr systems (i would love to add ldap and social auth mechanisms) but i need to know how to coordinate these things with you so i dont waste my time.  Please give some guidance here.

scphantm committed:
Merge branch 'develop' into feature/retro_add_tags

# Conflicts:
#	frontend/src/Settings/ImportLists/ImportLists/EditImportListModalContent.js

scphantm committed:
Updated retro add tags and added the bulk edit for retro tags

scphantm committed:
Merge branch 'Radarr:develop' into feature/retro_add_tags

scphantm committed:
lint fix

scphantm committed:
Refactor ImportListSyncService to adjust auto-import logic

scphantm committed:
Merge remote-tracking branch 'origin/feature/retro_add_tags' into feature/retro_add_tags

scphantm commented:
I was finally able to get back to this.  i did a develop merge to get everything updated and then went on to fix everything.  The one JS file that was being edited was already converted to TS so I was able to update that accordingly.  I updated the logic and got all the tests to run.  What else needs done to merge this in?

If you are wondering about use case, its pretty easy.  If I have 10 John Wayne movies already in my collection, and I create a user list with John Wayne.  In that user import list, I add a tag `john-wayne`.  If this is selected, when it adds new movies, it will tag them with `john-wayne`, and it will go back to the existing 10 movies in the collection and tags them with `john-wayne` as well.  When you are configured to generate `movie.nfo` files, those tags can be imported into Jellyfin and I believe Plex to be used in those applications.

This is also useful for Radarr.  I have over 10,000 movies in my library and probably around 14,000 in my Radarr collection.  I use import lists extensively and want to be able to pull up the list of movies that are list X.  Whether they were in my collection before I created the list or not.  

This would be immensely helpful because currently, the only way to pull up the movies in a list is via the discovery page, you can filter based on list.  But, you can't edit or see any info about the movies in this page.  I am constantly pulling up a list in discover, then right clicking on each movie in the discover page to open it in a new tab and make my edits, one movie at a time.  With this, I could update the list to retro tag the movies in that list with something, then simply filter based on the tag and bam.  I have all my movies that I can do all my movie maintenance things with.

I think this is very useful,

scphantm commented:
#2529 asked for the same feature all the way back in 2018, so this is not a new feature request.

scphantm committed:
Rename `retroApplyTags` to `tagExisting` across frontend and backend for consistent terminology.

scphantm commented:
In the process of merging this into Sonarr, it was suggested to refactor the feature from RetroApply to TagExisting.  I refactored this PR to match it.  It was a good suggestion.
